src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs

```diff
@@ -182,23 +194,49 @@ private async Task DoReceive()
                         // Pipe consumer is shut down, do we stop writing
                         break;
                     }
-                }
-            }
-            catch (SocketException ex) when (IsConnectionResetError(ex.SocketErrorCode))
-            {
-                // This could be ignored if _shutdownReason is already set.
-                error = new ConnectionResetException(ex.Message, ex);
 
-                // There's still a small chance that both DoReceive() and DoSend() can log the same connection reset.
-                // Both logs will have the same ConnectionId. I don't think it's worthwhile to lock just to avoid this.
-                if (!_socketDisposed)
-                {
-                    SocketsLog.ConnectionReset(_logger, this);
+                    bool IsNormalCompletion(TransferResult result)
+                    {
+                        if (!result.HasError)
+                        {
+                            return true;
+                        }
+
+                        if (IsConnectionResetError(result.SocketError!.SocketErrorCode))
+                        {
+                            // This could be ignored if _shutdownReason is already set.
+                            var ex = result.SocketError!;
+                            error = new ConnectionResetException(ex.Message, ex);
+
+                            // There's still a small chance that both DoReceive() and DoSend() can log the same connection reset.
+                            // Both logs will have the same ConnectionId. I don't think it's worthwhile to lock just to avoid this.
+                            if (!_socketDisposed)
+                            {
+                                SocketsLog.ConnectionReset(_logger, this);
+                            }
+
+                            return false;
+                        }
+
+                        if (IsConnectionAbortError(result.SocketError!.SocketErrorCode))
+                        {
+                            // This exception should always be ignored because _shutdownReason should be set.
+                            error = result.SocketError!; 
```

Comment:
Is this effectively treating socket errors that aren't considered either aborts or resets as "normal" completions? We definitely should throw for any socket error that we don't recognize.

Reply:
> We definitely should throw for any socket error that we don't recognize

It's still throwing the same exceptions outside of this component. This PR only removes any exceptions that were caught (expected) and not re-thrown.